source/client/service_main.h

```diff
@@ -19,18 +23,40 @@ class ServiceMain : public Envoy::Logger::Loggable<Envoy::Logger::Id::main> {
 public:
   ServiceMain(int argc, const char** argv);
   void start();
+  /**
+   * Can be used to wait for the server to exit.
```

Comment:
Can we explicitly say that this call is blocking?

Can we also indicate that it registers for certain signals that will terminate the server?